core/src/main/java/jenkins/security/QueueItemAuthenticatorMonitor.java

```diff
@@ -0,0 +1,140 @@
+    @Override
+    public boolean isActivated() {
+        AuthorizationStrategy authorizationStrategy = Jenkins.get().getAuthorizationStrategy();
+        if (authorizationStrategy instanceof AuthorizationStrategy.Unsecured) {
+            return false;
+        }
+        if (authorizationStrategy instanceof LegacyAuthorizationStrategy) {
+            return false;
+        }
+        if (authorizationStrategy instanceof FullControlOnceLoggedInAuthorizationStrategy) {
+            return false;
+        }
+        return !isQueueItemAuthenticatorPresent() || !isQueueItemAuthenticatorConfigured() || isAnyBuildLaunchedAsSystemWithAuthenticatorPresent();
+    }
+
+    @RequirePOST
+    public HttpResponse doAct(@QueryParameter String redirect, @QueryParameter String dismiss, @QueryParameter String reset) throws IOException {
+        if (redirect != null) {
+            return HttpResponses.redirectTo("https://jenkins.io/redirect/queue-item-security");
+        }
+        if (dismiss != null) {
+            this.disable(true);
+        }
+        if (reset != null) {
+            anyBuildLaunchedAsSystemWithAuthenticatorPresent = false;
+        }
+        return HttpResponses.forwardToPreviousPage();
+    }
+
+    public static boolean isQueueItemAuthenticatorPresent() {
+        // there is no QueueItemAuthenticatorDescriptor; perhaps you need to install authorize-project?
+        return !QueueItemAuthenticatorDescriptor.all().isEmpty();
+    }
+
+    public static boolean isQueueItemAuthenticatorConfigured() {
+        // there is no configured QueueItemAuthenticator; perhaps you need to configure it?
+        return !QueueItemAuthenticatorConfiguration.get().getAuthenticators().isEmpty();
+    }
+
+    public boolean isAnyBuildLaunchedAsSystemWithAuthenticatorPresent() {
+        // you configured a QueueItemAuthenticator, but builds are still running as SYSTEM
+        return anyBuildLaunchedAsSystemWithAuthenticatorPresent;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return Messages.QueueItemAuthenticatorMonitor_DisplayName();
+    }
+
+    @Restricted(NoExternalUse.class)
+    @Extension
+    public static class QueueListenerImpl extends QueueListener {
+        @Override
+        public void onLeft(Queue.LeftItem li) {
+            if (li.isCancelled()) {
+                return;
+            }
+
+            // XXX This can probably be done better, unsure what the expected priority order is
+            String displayName = li.getDisplayName();
+            if (displayName == null && li.task != null) {
+                displayName = li.task.getFullDisplayName();
+            }
+            Queue.Executable executable = li.getExecutable();
+            if (displayName == null && executable != null) {
+                displayName = executable.toString();
+            }
+
+            // TODO this is probably not intended to be used as a getter -- seems potentially unstable
```

Comment:
I think it is OK. `ExecutorChunk.canAccept` calls it, as does `Node.canTake`, but the main call is from `Executor.run`—and BTW I think [this](https://github.com/jenkinsci/jenkins/blob/dcf354ce7835beb10d9c5d7c1ff60f789734f4da/core/src/main/java/hudson/model/Executor.java#L419-L427) could be removed now.